[libc] stop including unistd.h in src/

nickdesaulniers opened the pull request:
It's not hermetic to include the system libc's headers and is one of the
reasons why our linter bot has been very red for a long time.

This is starting to be an issue for new contributors as well, since we're now
getting conflicts between the parts of our codebase that are hermetic, vs parts
that aren't.

I need to think about a rule of thumb or policy that can be publicly documented
so that we don't backslide.

I should probably clean up test/ as well, but let's clean up just src/ to
unblock fellow contributors.

Link: https://github.com/llvm/llvm-project/pull/85514#discussion_r1532078163

nickdesaulniers committed:
[libc] stop including unistd.h in src/

It's not hermetic to include the system libc's headers and is one of the
reasons why our linter bot has been very red for a long time.

This is starting to be an issue for new contributors as well, since we're now
getting conflicts between the parts of our codebase that are hermetic, vs parts
that aren't.

I need to think about a rule of thumb or policy that can be publicly documented
so that we don't backslide.

I should probably clean up test/ as well, but let's clean up just src/ to
unblock fellow contributors.

Link: https://github.com/llvm/llvm-project/pull/85514#discussion_r1532078163

nickdesaulniers committed:
missed one

$ grep -rn "#include <unistd" libc/src/

nickdesaulniers commented on libc/src/unistd/getopt.h:
```diff
@@ -9,8 +9,8 @@
-#include <stdio.h>
-#include <unistd.h>
+#include "include/llvm-libc-types/FILE.h"
+#include "src/__support/common.h"
```
not necessary

nickdesaulniers committed:
remove unnecessary include

michaelrj-google reviewed (commented):
You also need to update the bazel.

nickdesaulniers committed:
bazel support

nickdesaulniers committed:
fix libc_sys_mman_unittests

nickdesaulniers commented:
hmm...[a previous run](https://buildkite.com/llvm-project/github-pull-requests/builds/49030#018e5ce9-23c6-46aa-8713-8273405897c2) showed a failure running `ninja libc_unistd_unittests` for libc/test/src/unistd/sysconf_test.cpp. I wonder what could be going wrong there. I can't seem to reproduce locally.

nickdesaulniers commented:
> hmm...a previous run showed a failure running ninja libc_unistd_unittests for libc/test/src/unistd/sysconf_test.cpp. I wonder what could be going wrong there. I can't seem to reproduce locally.

Perhaps the inclusion of <unistd.h> in libc/test/src/unistd/sysconf_test.cpp is getting a different value for _SC_PAGESIZE than the one we provide.

We define this value to 1, but glibc's it's 30. Bionic defines this to 0x0027 (39).

nickdesaulniers committed:
fix unit tests using our _SC_PAGESIZE

This needs a more widespread fix, as we'll need to match definitions for
specific target platforms.

michaelrj-google reviewed (approved):
LGTM

michaelrj-google reviewed (changes requested):
This patch needs to be updated to use the new header indirection design.
